aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2287,10 +2288,15 @@ fn create_account_if_does_not_exist(
 pub(crate) fn is_account_init_for_sponsored_transaction(
     txn_data: &TransactionMetadata,
     features: &Features,
+    resolver: &impl AptosMoveResolver,
 ) -> bool {
     txn_data.fee_payer.is_some()
         && txn_data.sequence_number == 0
         && features.is_enabled(FeatureFlag::SPONSORED_AUTOMATIC_ACCOUNT_V1_CREATION)
+        && resolver
+            .get_resource(&txn_data.sender(), &AccountResource::struct_tag())
+            .map(|data| data.is_none())
+            .unwrap_or(true)
```

Comment:
Should we propagate the error when get_resource returns Err?

Reply:
I'm not sure it would ultimately have any impact other than an earlier flag that we have some bizarre invariant violation... your call.

Reply:
note, here we didn't return the error: https://github.com/aptos-labs/aptos-core/blob/da2db7e9c467884be5226a360fb1cf5499a13598/aptos-move/aptos-vm/src/gas.rs#L127

so I'd be hesitant to adopt this policy now.

Reply:
I spent more time thinking about this, and while returning none and error should be similar, if we can actually just exit early it requires less reasoning, hence I fixed this to return the error.